SmartDeviceLink/SDLSoftButtonManager.m

```diff
@@ -205,8 +205,9 @@ - (void)sdl_updateWithCompletionHandler:(nullable SDLSoftButtonUpdateCompletionH
     if (self.softButtonObjects == nil) {
         SDLLogV(@"Soft button objects are nil, sending an empty array");
         self.inProgressUpdate.softButtons = @[];
-    } else if (([self sdl_currentStateHasImages] && ![self sdl_allCurrentStateImagesAreUploaded])
-               && (self.softButtonCapabilities ? !self.softButtonCapabilities.imageSupported : YES)) {
+    } else if ((([self sdl_currentStateHasImages] && ![self sdl_allCurrentStateImagesAreUploaded])
+               && (self.softButtonCapabilities ? self.softButtonCapabilities.imageSupported.boolValue : NO))
+               || (self.softButtonCapabilities ? !self.softButtonCapabilities.imageSupported.boolValue : YES)){
```

Comment:
This if statement is super confusing, the parenthesis and newlines don't line up like you'd expect, there are too many parenthesis than are necessary `((A && B) && C)` == `(A && B && C)`, you use a BOOL value in one section and reverse check it in another. Please simplify it and pull out the `imagesSupported` BOOL into a local variable.

Also, I'm not sure this is correct, it seems to be saying (if i'm following the parenthesis right) that if:

1. The current state has images AND the images are not uploaded AND the system doesn't support images, send text buttons
2. If the head unit doesn't support images, send text buttons

Reply:
Fixed